environment-dev.yml

```diff
@@ -36,7 +36,7 @@ dependencies:
   - scikit-learn
   - pymssql
   - psycopg2
-  - mypy<0.990
+  - mypy<=0.991
```

Comment:
why do we still limit `mypy` version? The previous limitation to `<0.990` was introduced due to a buggy mypy release that broke our CI, since they released a new version with the fix I believe it's possible for us to relax mypy requirement for `mypy!=0.990`

Reply:
I created https://github.com/modin-project/modin/pull/5349